tests/test_undate.py

```diff
@@ -12,13 +12,110 @@ def test_str(self):
         assert str(Undate(2022)) == "2022"
         assert str(Undate(month=11, day=7)) == "--11-07"
 
+    def test_partially_known_str(self):
+        assert str(Undate("19XX")) == "19XX"
+        assert str(Undate(2022, "1X")) == "2022-1X"
+        assert str(Undate(2022, 11, "2X")) == "2022-11-2X"
+        assert str(Undate(month="1X", day=7)) == "--1X-07"
+
+        # TODO: should not allow initializing year/day without month;
+        # should we infer unknown month? or raise an exception?
+        # assert str(Undate(2022, day="2X")) == "2022-XX-2X"  # currently returns 2022-2X
+        # assert str(Undate(2022, day=7)) == "2022-XX-07"   @ currently returns 2022-07
```

Comment:
I could see a situation with a corrupted text where we legitimately only know the year and day of the month, but we can't do much with the additional day info in this case. I don't think it should error; it's also unlikely we can infer the month in this situation. Is it possible to bump to the known granularity (year) for calculation purposes, and leave the day in a string rendering of the `undate`? Can't remember